integration-tests/msa/createMsa.test.ts

```diff
@@ -1,69 +1,69 @@
 import "@frequency-chain/api-augment";
 import assert from "assert";
 import { ApiRx } from "@polkadot/api";
-import { connect, createKeys } from "../scaffolding/apiConnection"
-import { groupEventsByKey, signPayloadSr25519 } from "../scaffolding/helpers";
+import { connect } from "../scaffolding/apiConnection"
+import { AccountFundingInputs, createAccount, createAndFundAccount, generateFundingInputs, txAccountingHook, signPayloadSr25519 } from "../scaffolding/helpers";
 import { KeyringPair } from "@polkadot/keyring/types";
-import { addPublicKeyToMsa, createMsa, createSchema, deletePublicKey } from "../scaffolding/extrinsicHelpers";
+import { addPublicKeyToMsa, createMsa, createSchema, deletePublicKey, signAndSend } from "../scaffolding/extrinsicHelpers";
 import { AVRO_GRAPH_CHANGE } from "../schemas/fixtures/avroGraphChangeSchemaType";
-import { filter, firstValueFrom } from "rxjs";
 
-describe("Create Accounts", () => {
+describe("Create Accounts", function () {
+    this.timeout(15000);
+
+    let fundingInputs: AccountFundingInputs;
+
     let api: ApiRx;
-    let keys: KeyringPair;
 
-    before(async () => {
+    before(async function () {
         let connectApi = await connect(process.env.WS_PROVIDER_URL);
         api = connectApi
+        fundingInputs = generateFundingInputs(api, this.title);
     })
 
-    after(() => {
-        api.disconnect()
+    after(async function () {
+        await txAccountingHook(api, fundingInputs.context);
+        await api.disconnect()
     })
 
     // NOTE: We will need a sustainable way to create new keys for every test,
     // since there is only one node instance per test suite.
-    it("should successfully create an MSA account", async () => {
-        keys = createKeys("//Alice")
-        const chainEvents = await createMsa(api, keys)
+    it("should successfully create an MSA account", async function () {
+        const keys = (await createAndFundAccount(fundingInputs)).newAccount;
+        const chainEvents = await createMsa(api, keys);
 
-        assert.equal(chainEvents["system.ExtrinsicFailed"], undefined);
         assert.notEqual(chainEvents["system.ExtrinsicSuccess"], undefined);
         assert.notEqual(chainEvents["msa.MsaCreated"], undefined);
         assert.notEqual(chainEvents["transactionPayment.TransactionFeePaid"], undefined);
-    }).timeout(15000)
+    });
 
-    it("should successfully mimic a user's path using tokens", async () => {
-        keys = createKeys("//Charlie")
+    it("should successfully mimic a user's path using tokens", async function () {
```

Comment:
Curious. Does this only work if we switch it from async () => to async function()?

Reply:
I had to look this up as well. It's based on https://mochajs.org/#arrow-functions

That said @JoeCap08055 we might want to add a linter to enforce and autofix it.

Reply:
I'll look into adding a linter for the follow-on issue (see my comment below)